lib/index.ts

```diff
@@ -140,8 +192,68 @@ async function buildSubTreeRecursive(
     }
     // dependency was not found on a current path, remove last key (move closer to the root) and try again
     // visitedDepPaths can be passed by a reference, because traversing up doesn't update it
-    return buildSubTreeRecursive(depName, lockfilePath.slice(0, -1), lockFile, depPath);
+    return buildSubTreeRecursiveFromPackageLock(depName, lockfilePath.slice(0, -1), lockFile, depPath);
+  }
+}
+
+async function buildSubTreeRecursiveFromYarnLock(
+  searchedDep: Dep, lockFile: YarnLock, depPath: string[] ): Promise<PkgTree> {
+  const depSubTree: PkgTree = {
+    depType: searchedDep.dev ? DepType.dev : DepType.prod,
+    dependencies: {},
+    name: searchedDep.name,
+    version: undefined,
+  };
+
+  const depKey = `${searchedDep.name}@${searchedDep.version}`;
+
+  const dep = _.get(lockFile.object, depKey);
+
+  if (!dep) {
+    throw new Error(`Dependency ${depKey} was not found in yarn.lock.
+      Your package.json and yarn.lock are probably out of sync.
+      Please run npm install and try to parse the log again.`);
   }
+
+  if (depPath.indexOf(depKey) >= 0) {
+    depSubTree.cyclic = true;
+  } else {
+    depPath.push(depKey);
+    depSubTree.version = dep.version;
+    const newDeps = Object.entries({...dep.dependencies, ...dep.optionalDependencies});
+
+    await Promise.all(newDeps.map(async ([name, version]) => {
+      const newDep: Dep = {
+        dev: searchedDep.dev,
+        name,
+        version,
+      };
+      depSubTree.dependencies[name] = await buildSubTreeRecursiveFromYarnLock(
+        newDep, lockFile, []);
+    }));
+  }
+
+  return depSubTree;
+}
+
+function parseLockFile(lockFileRaw: string, lockfileType: LockfileType, includeDev: boolean): PackageLock | YarnLock {
+  let lockfile: PackageLock | YarnLock ;
+  switch (lockfileType) {
+    case LockfileType.npm:
+      lockfile = JSON.parse(lockFileRaw);
```

Comment:
What if the parsing fails? Do we have a test with an invalid json file? Would like to see us handle this with try/catch

Reply:
In progress :-)